src/BlockTypes/ProductQuery.php

```diff
@@ -133,31 +139,54 @@ public function build_query( $query ) {
 			function( $acc, $query ) {
 				return $this->merge_queries( $acc, $query );
 			},
-			$common_query_values
+			$base_query
```

Comment:
Here we merge the global query into queries by attributes and filters, which means that for example:

A Product Query block that inherits the query from the `Hoodies` category page can also display the products from the `Tshirts` category.

I don't think this is the correct behavior. We should honor the global query, we can add new query but shouldn't modify the query provided by the global query:
- On the product category page, we shouldn't modify the `tax_query`.
- But we can add filters, for example, showing only In Stock products from the current category.

Reply:
> but at the same time, we want that the user can use `Inherit query from template` with the `on-sale` attribute, right?

Yes, users can choose to display only on-sale products or only in-stock products on the Hoodies category page, but all products must be in the Hoodies category. The same applies to tag and product attribute archives.